b3sum/src/main.rs

```diff
@@ -118,6 +124,19 @@ fn write_output(mut output: blake3::OutputReader, mut len: u64) -> Result<()> {
     Ok(())
 }
 
+fn write_raw_output(mut output: blake3::OutputReader, mut len: u64) -> Result<()> {
+    let mut stdout = std::io::stdout();
+    let mut block = [0; blake3::BLOCK_LEN];
+    while len > 0 {
+        output.fill(&mut block);
+        let take_bytes = cmp::min(len, block.len() as u64);
+        stdout.write(&block[..take_bytes as usize])?;
+        len -= take_bytes;
+    }
```

Comment:
Rather than this loop, can we just use `std::io::copy`? Maybe together with `Read::take`? (A call to `lock()` above is probably still a good idea.)

Reply:
`std::io::copy` is unbounded, so we would just copying bytes infinitely since `blake3::OutputReader` never produces an `EOF`.  

I don't see any `std::io::copy_n` or equivalent that is limited to `n` bytes.    This would be a nice edition to the standard library honestly.

Reply:
I just learned about `Reader::take()` so this is now fixed.